spec/controllers/saml_idp_controller_spec.rb

```diff
@@ -1120,6 +1122,56 @@ def name_id_version(format_urn)
       end
     end
 
+    context 'no matching cert from the SAML request' do
+      let(:user) { create(:user, :signed_up) }
+
+      before do
+        stub_analytics
+        allow(@analytics).to receive(:track_event)
+      end
+
+      it 'notes that in the analytics event' do
+        auth_settings = saml_settings(
+          overrides: {
+            authn_context: [
+              Saml::Idp::Constants::IAL1_AUTHN_CONTEXT_CLASSREF,
+              Saml::Idp::Constants::DEFAULT_AAL_AUTHN_CONTEXT_CLASSREF,
+            ],
+            security: {
```

Comment:
This turns off the SAML assertion signature for the `AuthnRequest`, so there can be no matching cert. I didn't write an explicit test for when the signature is invalid or uses a cert that isn't configured, but the result should be the same.